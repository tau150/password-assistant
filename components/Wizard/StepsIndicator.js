import * as React from 'react'
import PropTypes from 'prop-types'
import { Badge, Box, Flex } from '@chakra-ui/react'

function StepsIndicator({stepsNumber, activeStep}){
  const arrayOfSteps = Array.from(Array(stepsNumber), (_, idx) => idx + 1)

  return (
    <Flex alignItems='center' position='relative'>
      {arrayOfSteps.map(number => (
        <Box key={number} position='relative'>
          {number !== stepsNumber && <Box left='20px' top='calc(50% - 2.5px)' zIndex='-1' position='absolute' as='span' w='100px' h='5px' bgColor='brand.accent'/>}
          <Badge
            fontSize='18px'
            w={activeStep === number ? "40px" : '30px'}
            h={activeStep === number ? "40px" : '30px'}
            borderRadius='50%'
            d='flex'
            justifyContent='center'
            alignItems='center'
            color='white'
            bgColor={activeStep === number ? "brand.secondary" : 'brand.accent'}
            boxShadow={activeStep === number ? "lg" : 'initial'}
            role="status"
            aria-label={`${number} ${activeStep === number ? 'is active' : ''}`}
            m='20px'
            position='relative'
            zIndex={3}
            key={number}>
              {number}
          </Badge>
        </Box>))}

    </Flex>
  )
}

StepsIndicator.propTypes = {
  stepsNumber: PropTypes.number,
  activeStep: PropTypes.number
}

export default StepsIndicator