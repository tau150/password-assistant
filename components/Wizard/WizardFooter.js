import * as React from 'react'
import PropTypes from 'prop-types'
import { FormattedMessage } from 'react-intl'
import { Flex } from '@chakra-ui/react'

import Button from '../ui/Button'

function WizardFooter({ handlePrevStep, handleNextStep, isStepDisabled, isPrevDisabled}){
  return (
    <Flex as='footer' borderTop='1px' py={12} px={20} borderColor='gray.200' alignItems='center' justifyContent='space-between'>
      <Button aria-label='prev-button' variant="secondary" disabled={isPrevDisabled} onClick={handlePrevStep}>
        <FormattedMessage  id='cancel' defaultMessage='Cancel'/>
      </Button>
      <Button aria-label='next-button' variant="primary" disabled={isStepDisabled} onClick={handleNextStep}>
        <FormattedMessage id='next' defaultMessage='Next'/>
      </Button>
    </Flex>
  )
}

WizardFooter.propTypes = {
  handlePrevStep: PropTypes.func.isRequired,
  handleNextStep: PropTypes.func.isRequired,
  isStepDisabled: PropTypes.bool,
  isPrevDisabled: PropTypes.bool
}

export default WizardFooter