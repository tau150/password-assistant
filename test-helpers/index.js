
import * as React from 'react'
import {   render as rtlRender } from '@testing-library/react'
import { IntlProvider } from 'react-intl';
import { ChakraProvider } from "@chakra-ui/react"

import theme from '../styles/theme';
import intlENmessages from '../content/locale/en.json'

function AppProviders({children}){
  return (
    <IntlProvider
      locale='en'
      messages={intlENmessages}
      onError={() => null}>
      <ChakraProvider theme={theme}>
        {children}
      </ChakraProvider>
    </IntlProvider>
  )
}


export function render(ui, ...renderOptions){
  const returnValue = {
    ...rtlRender(ui, {wrapper: AppProviders, ...renderOptions}),
  }
  return returnValue
}
