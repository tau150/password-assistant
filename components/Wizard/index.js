import * as React from 'react'
import PropTypes from 'prop-types'
import { Box, Flex } from '@chakra-ui/react'

import { StepWizardContext } from '../../contexts/stepWizardContext'
import StepsIndicator from './StepsIndicator'
function Wizard({children}){
  const [currentStep, setCurrentStep] = React.useState(0)
  const [isStepDisabled, setIsStepDisabled] = React.useState(false)

  const stepsNumberBasedOnChildren = React.Children.count(children);

  const handleNextStep = async () => {
    setCurrentStep( prev => prev += 1)
  }

  const handlePrevStep = async () => {
    setCurrentStep( prev => prev -= 1)
  }

  const arrayOfChildren = React.Children.toArray(children);
  const currentContent = arrayOfChildren[currentStep];
  const isLastStep = currentStep + 1 === stepsNumberBasedOnChildren

  const value = {
    currentStep,
    numberOfSteps: stepsNumberBasedOnChildren,
    setCurrentStep,
    setIsStepDisabled,
    isStepDisabled,
    handleNextStep,
    handlePrevStep,
    isLastStep
  }

  return (
    <StepWizardContext.Provider value={value}>
      <Box border='1px solid brand.accent' boxShadow='lg'>
        <Flex as='header' position='relative' zIndex='2' h='80px' bgColor='gray.200' alignItems='center' justifyContent='center' boxShadow='md'>
          <StepsIndicator stepsNumber={stepsNumberBasedOnChildren} activeStep={currentStep + 1}/>
        </Flex>
        <Box p='4'>
          {currentContent}
        </Box>
      </Box>
    </StepWizardContext.Provider>
  )
}


Wizard.propTypes = {
  children: PropTypes.node,
}

export default Wizard