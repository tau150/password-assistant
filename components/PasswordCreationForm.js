import * as React from 'react'
import PropTypes from 'prop-types'
import { Box, Heading, Text, Stack, Spinner, Flex} from '@chakra-ui/react'
import { FormattedMessage, useIntl } from 'react-intl'

import { submitForm } from '../pages/api/open'
import { status as statusConst } from '../constants/ui'
import { useStepsWizard } from '../contexts/stepWizardContext'
import { validationKeys } from '../constants/validations'
import WizardFooter from './Wizard/WizardFooter'
import Input from './ui/Input'


const passwordValidations = [
  {type: validationKeys.MIN, reference: 8},
  {type: validationKeys.MAX, reference: 24},
  {type: validationKeys.ONE_UPPER },
  {type: validationKeys.ONE_SYMBOL }
]

function PasswordCreationForm({setPasswordResponse}){
  const intl = useIntl()
  const { setIsStepDisabled } = useStepsWizard()


  const [,setClueInputValue] = React.useState('')
  const [passWordInputValue, setPasswordInputValue] = React.useState('')
  const [,setPasswordConfInputValue] = React.useState('')
  const [isPasswordValid, setIsPasswordValid] = React.useState(false)
  const [isPasswordConfValid, setIsPasswordConfValid] = React.useState(false)
  const [isLoading, setIsLoading] = React.useState(false)


  React.useEffect( () => {
    setIsStepDisabled(!isPasswordValid || !isPasswordConfValid)
  }, [isPasswordValid,isPasswordConfValid])


  const handleChangeClueInput = (e) => {
    const { value } = e.target
    if(value.length < 61){
      setClueInputValue(value)
    }
  }

  const handleChangePasswordInput = (e, isValid) => {
    const { value } = e.target
    setPasswordInputValue(value)
    setIsPasswordValid(isValid)
  }

  const handleChangePasswordConfInput = (e, isValid) => {
    const { value } = e.target
    setPasswordConfInputValue(value)
    setIsPasswordConfValid(isValid)
  }

  const onClickNext = async () => {
    setIsLoading(true)
    try {
     const res = await submitForm(passWordInputValue)
     const { status } = res
     if(status === 200){
      setPasswordResponse(statusConst.SUCCESS)
     }
    }catch(e){
      setPasswordResponse(statusConst.ERROR)
    }
  }

  return (
    <Box as='section' p={18} mt={12}>
      {isLoading ? (
        <Flex h='400px' alignItems='center' justifyContent='center'>
        <Spinner
          thickness="4px"
          speed="0.65s"
          emptyColor="gray.200"
          color="blue.500"
          size="xl"
        />
      </Flex>
      ) : (
      <>
        <Heading as='h2' fontSize='3xl'>
          <FormattedMessage id='step1Title' defaultMessage='Create your Password Manager'/>
        </Heading>
        <Text maxW={{xs: '100%', md: '50%'}} mt={8}>
          <FormattedMessage
            id='howWorksDesc'
            defaultMessage='First, you must create a different password for your electronic belongings. You will not be able to recover your password, so remember it well.'
          />
        </Text>
        <form>
          <Stack mt={12} direction={["column", "row"]} spacing={['140px', 20]}>
            <Box w={['100%', '30%']}>
              <Input
                shouldToggleShow={true}
                placeholder={intl.formatMessage({ id: 'createPasswordLabel', defaultMessage: 'Create your Master Password' })}
                name='password'
                id='password'
                label={intl.formatMessage({ id: 'createPasswordLabel', defaultMessage: 'Create your Master Password' })}
                handleOnChange={handleChangePasswordInput}
                validations={passwordValidations}
              />
            </Box>
            <Box w={['100%', '30%']}>
              <Input
                shouldToggleShow={true}
                placeholder={intl.formatMessage({ id: 'confirmPasswordLabel', defaultMessage: 'Repeat your Master Password' })}
                name='passwordConfirmation'
                id='passwordConfirmation'
                label={intl.formatMessage({ id: 'confirmPasswordLabel', defaultMessage: 'Repeat your Master Password' })}
                handleOnChange={handleChangePasswordConfInput}
                validations={[{type: validationKeys.EQUAL, reference: passWordInputValue}]}
              />
            </Box>
          </Stack>
          <Box w={['100%', '65%']} mt={36}>
            <Input
              placeholder={intl.formatMessage({ id: 'confirmPasswordLabel', defaultMessage: 'Repeat your Master Password' })}
              name='clue'
              id='clue'
              shouldCountChars={true}
              label={intl.formatMessage({ id: 'helpPasswordLabel', defaultMessage: 'Create your clue to remember your password (optional)' })}
              tooltipText={intl.formatMessage({ id: 'helpPasswordInfo', defaultMessage: 'This will help you to remember your password' })}
              ariaTooltip={intl.formatMessage({ id: 'rememberAria', defaultMessage: 'Remember Password' })}
              placeholder={intl.formatMessage({ id: 'helpPasswordPh', defaultMessage: 'Enter a clue' })}
              handleOnChange={handleChangeClueInput}
            />
          </Box>
        </form>
      </>
      )}
      <Box mt={8}>
        <WizardFooter onClickNext={onClickNext}/>
      </Box>
    </Box>
  )
}


PasswordCreationForm.propTypes = {
  setPasswordResponse: PropTypes.func
}

export default PasswordCreationForm