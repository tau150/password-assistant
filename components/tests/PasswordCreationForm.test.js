import * as React from 'react'
import { render } from '../../test-helpers'

import PasswordCreationForm from '../PasswordCreationForm'
import Wizard from '../Wizard'


function Wrapper() {
  return (
    <Wizard>
      <PasswordCreationForm />
    </Wizard>
  )
}

describe('Password Confirmation Form test', () => {
  it('renders Password Confirmation', () => {
    expect(render(<Wrapper />)).toMatchSnapshot()
  })

})

