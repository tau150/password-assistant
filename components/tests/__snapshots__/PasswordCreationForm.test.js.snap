// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Password Confirmation Form test renders Password Confirmation 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body
    class="chakra-ui-light"
  >
    <div
      id="chakra-toast-portal"
    >
      <ul
        id="chakra-toast-manager-top"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; margin: 0px auto;"
      />
      <ul
        id="chakra-toast-manager-top-left"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
      <ul
        id="chakra-toast-manager-top-right"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
      <ul
        id="chakra-toast-manager-bottom-left"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
      <ul
        id="chakra-toast-manager-bottom"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; margin: 0px auto;"
      />
      <ul
        id="chakra-toast-manager-bottom-right"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
    </div>
    <div>
      <div
        class="css-l50gwe"
      >
        <header
          class="css-62ow20"
        >
          <div
            class="css-14s4iyf"
          >
            <div
              class="css-79elbk"
            >
              <span
                aria-label="1 is active"
                class="chakra-badge css-1aiadu4"
                role="status"
              >
                1
              </span>
            </div>
          </div>
        </header>
        <div
          class="css-h94677"
        >
          <section
            class="css-1vf38yy"
          >
            <h2
              class="chakra-heading css-1n7pa12"
            >
              Create your Password Manager
            </h2>
            <p
              class="chakra-text css-sula1c"
            >
              First, you must create a different password for your electronic belongings. You will not be able to recover your password, so remember it well.
            </p>
            <form>
              <div
                class="chakra-stack css-10bun6e"
              >
                <div
                  class="css-q7cxmg"
                >
                  <div
                    class="chakra-form-control css-1kxonj9"
                    role="group"
                  >
                    <label
                      class="chakra-form__label css-1bpd1fc"
                      for="password"
                      id="field-1-label"
                    >
                      Create your Master Password
                    </label>
                    <div
                      class="chakra-input__group css-4302v8"
                    >
                      <input
                        aria-label="password"
                        class="chakra-input css-155nkyn"
                        id="password"
                        name="password"
                        placeholder="Create your Master Password"
                        role="input"
                        type="password"
                        value=""
                      />
                      <div
                        class="chakra-input__right-element css-ohd79a"
                      >
                        <svg
                          class="chakra-icon css-13otjrl"
                          data-testid="eye-on"
                          fill="none"
                          focusable="false"
                          height="24"
                          stroke="currentColor"
                          stroke-linecap="round"
                          stroke-linejoin="round"
                          stroke-width="2"
                          viewBox="0 0 24 24"
                          width="24"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M1 12s4-8 11-8 11 8 11 8-4 8-11 8-11-8-11-8z"
                          />
                          <circle
                            cx="12"
                            cy="12"
                            r="3"
                          />
                        </svg>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="css-q7cxmg"
                >
                  <div
                    class="chakra-form-control css-1kxonj9"
                    role="group"
                  >
                    <label
                      class="chakra-form__label css-1bpd1fc"
                      for="passwordConfirmation"
                      id="field-2-label"
                    >
                      Repeat your Master Password
                    </label>
                    <div
                      class="chakra-input__group css-4302v8"
                    >
                      <input
                        aria-label="passwordConfirmation"
                        class="chakra-input css-155nkyn"
                        id="passwordConfirmation"
                        name="passwordConfirmation"
                        placeholder="Repeat your Master Password"
                        role="input"
                        type="password"
                        value=""
                      />
                      <div
                        class="chakra-input__right-element css-ohd79a"
                      >
                        <svg
                          class="chakra-icon css-13otjrl"
                          data-testid="eye-on"
                          fill="none"
                          focusable="false"
                          height="24"
                          stroke="currentColor"
                          stroke-linecap="round"
                          stroke-linejoin="round"
                          stroke-width="2"
                          viewBox="0 0 24 24"
                          width="24"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M1 12s4-8 11-8 11 8 11 8-4 8-11 8-11-8-11-8z"
                          />
                          <circle
                            cx="12"
                            cy="12"
                            r="3"
                          />
                        </svg>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="css-1mmvxjv"
              >
                <div
                  class="chakra-form-control css-1kxonj9"
                  role="group"
                >
                  <label
                    class="chakra-form__label css-1bpd1fc"
                    for="clue"
                    id="field-3-label"
                  >
                    Create your clue to remember your password (optional)
                    <svg
                      aria-label="Remember Password"
                      class="chakra-icon css-ehdjs0"
                      fill="none"
                      focusable="false"
                      height="24"
                      role="tooltip"
                      stroke="currentColor"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      stroke-width="2"
                      viewBox="0 0 24 24"
                      width="24"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <circle
                        cx="12"
                        cy="12"
                        r="10"
                      />
                      <line
                        x1="12"
                        x2="12"
                        y1="16"
                        y2="12"
                      />
                      <line
                        x1="12"
                        x2="12.01"
                        y1="8"
                        y2="8"
                      />
                    </svg>
                  </label>
                  <div
                    class="chakra-input__group css-4302v8"
                  >
                    <input
                      aria-describedby="field-3-helptext"
                      aria-label="clue"
                      class="chakra-input css-155nkyn"
                      id="clue"
                      name="clue"
                      placeholder="Enter a clue"
                      role="input"
                      type="text"
                      value=""
                    />
                    <div
                      class="chakra-input__right-element css-ohd79a"
                    />
                  </div>
                  <div
                    class="chakra-form__helper-text css-19qd4o4"
                    id="field-3-helptext"
                  >
                    0/60
                  </div>
                </div>
              </div>
            </form>
            <div
              class="css-2qrmgs"
            >
              <footer
                class="css-1ry2npb"
              >
                <button
                  aria-label="prev-button"
                  class="chakra-button css-1t0rcyz"
                  disabled=""
                  type="button"
                >
                  Cancel
                </button>
                <button
                  aria-label="next-button"
                  class="chakra-button css-1t0rcyz"
                  disabled=""
                  type="button"
                >
                  Next
                </button>
              </footer>
            </div>
          </section>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="css-l50gwe"
    >
      <header
        class="css-62ow20"
      >
        <div
          class="css-14s4iyf"
        >
          <div
            class="css-79elbk"
          >
            <span
              aria-label="1 is active"
              class="chakra-badge css-1aiadu4"
              role="status"
            >
              1
            </span>
          </div>
        </div>
      </header>
      <div
        class="css-h94677"
      >
        <section
          class="css-1vf38yy"
        >
          <h2
            class="chakra-heading css-1n7pa12"
          >
            Create your Password Manager
          </h2>
          <p
            class="chakra-text css-sula1c"
          >
            First, you must create a different password for your electronic belongings. You will not be able to recover your password, so remember it well.
          </p>
          <form>
            <div
              class="chakra-stack css-10bun6e"
            >
              <div
                class="css-q7cxmg"
              >
                <div
                  class="chakra-form-control css-1kxonj9"
                  role="group"
                >
                  <label
                    class="chakra-form__label css-1bpd1fc"
                    for="password"
                    id="field-1-label"
                  >
                    Create your Master Password
                  </label>
                  <div
                    class="chakra-input__group css-4302v8"
                  >
                    <input
                      aria-label="password"
                      class="chakra-input css-155nkyn"
                      id="password"
                      name="password"
                      placeholder="Create your Master Password"
                      role="input"
                      type="password"
                      value=""
                    />
                    <div
                      class="chakra-input__right-element css-ohd79a"
                    >
                      <svg
                        class="chakra-icon css-13otjrl"
                        data-testid="eye-on"
                        fill="none"
                        focusable="false"
                        height="24"
                        stroke="currentColor"
                        stroke-linecap="round"
                        stroke-linejoin="round"
                        stroke-width="2"
                        viewBox="0 0 24 24"
                        width="24"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M1 12s4-8 11-8 11 8 11 8-4 8-11 8-11-8-11-8z"
                        />
                        <circle
                          cx="12"
                          cy="12"
                          r="3"
                        />
                      </svg>
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="css-q7cxmg"
              >
                <div
                  class="chakra-form-control css-1kxonj9"
                  role="group"
                >
                  <label
                    class="chakra-form__label css-1bpd1fc"
                    for="passwordConfirmation"
                    id="field-2-label"
                  >
                    Repeat your Master Password
                  </label>
                  <div
                    class="chakra-input__group css-4302v8"
                  >
                    <input
                      aria-label="passwordConfirmation"
                      class="chakra-input css-155nkyn"
                      id="passwordConfirmation"
                      name="passwordConfirmation"
                      placeholder="Repeat your Master Password"
                      role="input"
                      type="password"
                      value=""
                    />
                    <div
                      class="chakra-input__right-element css-ohd79a"
                    >
                      <svg
                        class="chakra-icon css-13otjrl"
                        data-testid="eye-on"
                        fill="none"
                        focusable="false"
                        height="24"
                        stroke="currentColor"
                        stroke-linecap="round"
                        stroke-linejoin="round"
                        stroke-width="2"
                        viewBox="0 0 24 24"
                        width="24"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M1 12s4-8 11-8 11 8 11 8-4 8-11 8-11-8-11-8z"
                        />
                        <circle
                          cx="12"
                          cy="12"
                          r="3"
                        />
                      </svg>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="css-1mmvxjv"
            >
              <div
                class="chakra-form-control css-1kxonj9"
                role="group"
              >
                <label
                  class="chakra-form__label css-1bpd1fc"
                  for="clue"
                  id="field-3-label"
                >
                  Create your clue to remember your password (optional)
                  <svg
                    aria-label="Remember Password"
                    class="chakra-icon css-ehdjs0"
                    fill="none"
                    focusable="false"
                    height="24"
                    role="tooltip"
                    stroke="currentColor"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="2"
                    viewBox="0 0 24 24"
                    width="24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <circle
                      cx="12"
                      cy="12"
                      r="10"
                    />
                    <line
                      x1="12"
                      x2="12"
                      y1="16"
                      y2="12"
                    />
                    <line
                      x1="12"
                      x2="12.01"
                      y1="8"
                      y2="8"
                    />
                  </svg>
                </label>
                <div
                  class="chakra-input__group css-4302v8"
                >
                  <input
                    aria-describedby="field-3-helptext"
                    aria-label="clue"
                    class="chakra-input css-155nkyn"
                    id="clue"
                    name="clue"
                    placeholder="Enter a clue"
                    role="input"
                    type="text"
                    value=""
                  />
                  <div
                    class="chakra-input__right-element css-ohd79a"
                  />
                </div>
                <div
                  class="chakra-form__helper-text css-19qd4o4"
                  id="field-3-helptext"
                >
                  0/60
                </div>
              </div>
            </div>
          </form>
          <div
            class="css-2qrmgs"
          >
            <footer
              class="css-1ry2npb"
            >
              <button
                aria-label="prev-button"
                class="chakra-button css-1t0rcyz"
                disabled=""
                type="button"
              >
                Cancel
              </button>
              <button
                aria-label="next-button"
                class="chakra-button css-1t0rcyz"
                disabled=""
                type="button"
              >
                Next
              </button>
            </footer>
          </div>
        </section>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
